nestkernel/connection_manager.cpp

```diff
@@ -1089,6 +1089,21 @@ nest::ConnectionManager::get_connections( const DictionaryDatum& params ) const
     target_a = dynamic_cast< TokenArray const* >( target_t.datum() );
   }
 
+  // if connections have changed, (re-)build presynaptic infrastructure,
```

Comment:
Can we comment in the code on why this is in this get_connections() rather than the other get_connections()? The best case would be a description here fully specifying what the distinction between the two are, and why there are functional difference between the two, rather than simply a different signature that gets wrapped to a common get_connections() call that handles the functional parts.

Reply:
@hakonsbm as far as I remember, you moved this to fix the deadlock, right? Could you add a short explanation here?

Reply:
@apeyser Building the presynaptic infrastructure requires some MPI-communication. That caused problems when it was in the other `get_connections()`, as that one may return early when the number of connections for the given synapse model is zero. Therefore we moved it here. Additionally, the presynaptic infrastructure only has to be rebuilt once, not for every synapse model.